src/parquet/column/properties.h

```diff
@@ -126,26 +133,87 @@ class PARQUET_EXPORT WriterProperties {
       return this;
     }
 
-    Builder* encoding(
-        const std::shared_ptr<schema::ColumnPath>& path, Encoding::type encoding_type) {
-      return encoding(path->ToDotString(), encoding_type);
+    Builder* encodings(const std::shared_ptr<schema::ColumnPath>& path,
+        std::vector<Encoding::type>& encoding_type) {
+      return encodings(path->ToDotString(), encoding_type);
     }
 
-    Builder* encoding(const std::string& column_path, Encoding::type encoding_type) {
+    Builder* encodings(
+        const std::string& column_path, std::vector<Encoding::type>& encoding_type) {
       encodings_[column_path] = encoding_type;
       return this;
     }
 
-    Builder* encoding(Encoding::type encoding_type) {
+    Builder* value_encoding(
+        const std::shared_ptr<schema::ColumnPath>& path, Encoding::type encoding_type) {
+      return value_encoding(path->ToDotString(), encoding_type);
+    }
+
+    Builder* value_encoding(
+        const std::string& column_path, Encoding::type encoding_type) {
+      std::vector<Encoding::type> encodings(3);
+      encodings_[column_path] = encodings;
+      encodings_[column_path][2] = encoding_type;
+      encodings_[column_path][1] = default_dictionary_encoding_;
+      encodings_[column_path][0] = default_level_encoding_;
+      return this;
+    }
+
+    Builder* value_encoding(Encoding::type encoding_type) {
       default_encoding_ = encoding_type;
       return this;
     }
 
+    Builder* dictionary_encoding(
+        const std::shared_ptr<schema::ColumnPath>& path, Encoding::type encoding_type) {
+      return dictionary_encoding(path->ToDotString(), encoding_type);
+    }
+
+    Builder* dictionary_encoding(
+        const std::string& column_path, Encoding::type encoding_type) {
+      std::vector<Encoding::type> encodings(3);
+      encodings_[column_path] = encodings;
+      encodings_[column_path][2] = default_encoding_;
+      encodings_[column_path][1] = encoding_type;
+      encodings_[column_path][0] = default_level_encoding_;
+      return this;
+    }
+
+    Builder* dictionary_encoding(Encoding::type encoding_type) {
+      default_dictionary_encoding_ = encoding_type;
+      return this;
+    }
+
+    Builder* level_encoding(
+        const std::shared_ptr<schema::ColumnPath>& path, Encoding::type encoding_type) {
+      return level_encoding(path->ToDotString(), encoding_type);
+    }
+
+    Builder* level_encoding(
+        const std::string& column_path, Encoding::type encoding_type) {
+      std::vector<Encoding::type> encodings(3);
+      encodings_[column_path] = encodings;
+      encodings_[column_path][2] = default_encoding_;
+      encodings_[column_path][1] = default_dictionary_encoding_;
+      encodings_[column_path][0] = encoding_type;
```

Comment:
I don't think dictionary encoding should be a separate method. Both in V1 and V2, dictionary are only PLAIN encoded. We should hard-code like parquet-mr https://github.com/Parquet/parquet-mr/blob/fa8957d7939b59e8d391fa17000b34e865de015d/parquet-column/src/main/java/parquet/column/ParquetProperties.java